tests/trainer/test_trainer.py

```diff
@@ -501,7 +501,7 @@ def test_benchmark_option(tmpdir):
 
     # fit model
     trainer = Trainer(
-        default_root_dir=tmpdir,
+        default_root_dir=str(tmpdir),
```

Comment:
rather so this casting in Trainer

Reply:
Sounds good. will make that change.

Reply:
The reason to do it here is that the trainer and other places _correctly_ type annotate that it should be a `str` or `Optional[str]` and it is hte tests which are wrong in passing a path object. @Borda I think if the tests are going to do this we should update the type annotations to accept a union of string and path. what do you think?

Reply:
yes, then pls update annotation :]